desktop/cmp/rest/RestForm.js

```diff
@@ -47,29 +49,26 @@ export class RestForm extends Component {
             mask({model: model.loadModel, spinner: true})
         ];
     }
-    
+
     getForm() {
+        const {isWritable} = this.model;
         return vframe(
-            this.model.controlModels.map((model, idx) => {
-                return idx == 0 ? restControl({model, autoFocus: true}) : restControl({model});
-            })
+            this.model.controlModels.map((model, idx) => restControl({model, disabled: !isWritable, autoFocus: idx === 0}))
         );
     }
 
     getButtons() {
-        const {isValid, isWritable, isDirty, isAdd, actionEnabled} = this.model;
+        const {isValid, isWritable, isDirty, record, toolbarActions} = this.model;
 
         return [
-            button({
-                text: 'Delete',
-                icon: Icon.delete(),
-                intent: 'danger',
-                onClick: this.onDeleteClick,
-                omit: !actionEnabled.del || isAdd
+            recordActionBar({
+                actions: toolbarActions,
+                actionMetadata: {restModel: this.model},
```

Comment:
thought the plan was to have the actions all passed their grid, and to have the model hanging off of the grid?  Still find the passing around of the actionMetadata confusing.